### PR TITLE
ISPN-8024 Local Spring Infinispan Hibernate cache tutorial

### PR DESCRIPTION
**Pull request by galderz**
Part of http://issues.jboss.org/browse/ISPN-8024

It uses 9.2.0-SNAPSHOT until next release of 9.2.x where two Hibernate Cache providers have been implemented for Hibernate 5.2 and 5.1 respectively. This demo uses Hibernate 5.2, so it uses the correct cache provider for that version.

### CODE REVIEW

File: hibernate-cache/spring-local/src/main/java/org/infinispan/tutorial/simple/hibernate/cache/spring/local/InfinispanHibernateCacheSpringLocal.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+@SpringBootApplication
+public class InfinispanHibernateCacheSpringLocal {
+
+   private static final Logger log = LoggerFactory.getLogger(InfinispanHibernateCacheSpringLocal.class);
+
+   @Autowired
+   private EntityManagerFactory emf;
+
+   @Bean
+   public CommandLineRunner demo() {
```
Comment (slaskawi) [inline hibernate-cache/spring-local/src/main/java/org/infinispan/tutorial/simple/hibernate/cache/spring/local/InfinispanHibernateCacheSpringLocal.java]: For me personally, this tutorial is just too complicated. I would prefer to put it into the demos and replace the code here with something really simple, like putting an entity using `EntityManager` and obtaining it from Infinispan (or just check if it's there).

File: hibernate-cache/spring-local/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Enable those entities marked with @Cacheable to be cached
+spring.jpa.properties.javax.persistence.sharedCache.mode=ENABLE_SELECTIVE
+# Enables second level cache
+spring.jpa.properties.hibernate.cache.use_second_level_cache=true
+# Enable query cache
+spring.jpa.properties.hibernate.cache.use_query_cache=true
+
+# Use Infinispan second level cache provider
+spring.jpa.properties.hibernate.cache.region.factory_class=infinispan
```
Comment (slaskawi) [inline hibernate-cache/spring-local/src/main/resources/application.properties]: I'm pretty sure some of those are set by default.

Comment (galderz) [reply hibernate-cache/spring-local/src/main/resources/application.properties]: None of those are default.

File: hibernate-cache/spring-local/src/main/java/org/infinispan/tutorial/simple/hibernate/cache/spring/local/InfinispanHibernateCacheSpringLocal.java (hunk 1 above)
Comment (galderz) [reply hibernate-cache/spring-local/src/main/java/org/infinispan/tutorial/simple/hibernate/cache/spring/local/InfinispanHibernateCacheSpringLocal.java]: Hibernate does not expose internal Infinispan details to users. There's no way to achieve that and we don't even want to encourage that.

I think the tutorial is thorough but not complicated. It deals with very simple/common things for Hibernate users: entities, queries and caching concepts (such as expiring data from the cache)

**Comment by galderz** [comment]
@slaskawi I'm integrating this (this is now being linked to from docs since CR1). Feel free to make more comments or to reply to mine and we can enhance it further.

**Comment by galderz** [comment]
Don't worry about snapshot dependency, updating to CR1 as we speak.

**Pull request merged**